app/sidekiq/benefits_intake_status_job.rb

```diff
@@ -130,4 +117,10 @@ def log_result(result, form_id, uuid, time_to_transition = nil, error_message =
       Rails.logger.info('BenefitsIntakeStatusJob', result:, form_id:, uuid:, time_to_transition:)
     end
   end
+
+  def form_submission_attempts_hash
+    @_form_submission_attempts_hash ||= FormSubmissionAttempt
+                                        .where(aasm_state: 'pending')
+                                        .index_by(&:benefits_intake_uuid)
+  end
```

Comment:
How many 'pending' form submission attempts would we have at a given run? If it's a lot, this could be a somewhat memory intensive process given we're indexing them all.

Reply:
True. This is the alternative to looping through them all multiple times per run so I think it's ultimately worthwhile. @wayne-weibel suggested indexing them up front, above in this PR chatter and I think it's a good idea. Are there other ways to do this even more efficiently?